test/adapter/json_api/parse_test.rb

```diff
@@ -0,0 +1,73 @@
+require 'test_helper'
+module ActiveModel
+  class Serializer
+    module Adapter
+      class JsonApi
+        module Deserialization
+          class ParseTest < Minitest::Test
+            def setup
+              @hash = {
+                'data' => {
+                  'type' => 'photos',
+                  'id' => 'zorglub',
+                  'attributes' => {
+                    'title' => 'Ember Hamster',
+                    'src' => 'http://example.com/images/productivity.png'
+                  },
+                  'relationships' => {
+                    'author' => {
+                      'data' => nil
+                    },
+                    'photographer' => {
+                      'data' => { 'type' => 'people', 'id' => '9' }
+                    },
+                    'comments' => {
+                      'data' => [
+                        { 'type' => 'comments', 'id' => '1' },
+                        { 'type' => 'comments', 'id' => '2' }
+                      ]
+                    }
+                  }
+                }
+              }
+              @expected = {
+                id: 'zorglub',
+                title: 'Ember Hamster',
+                src: 'http://example.com/images/productivity.png',
+                author_id: nil,
+                photographer_id: '9',
+                comment_ids: %w(1 2)
+              }
+            end
+
+            def test_hash
+              parsed_hash = ActiveModel::Serializer::Adapter::JsonApi::Deserialization.parse(@hash)
+              assert_equal(@expected, parsed_hash)
+            end
+
+            def test_parameters
+              parameters = ActionController::Parameters.new(@hash)
+              parsed_hash = ActiveModel::Serializer::Adapter::JsonApi::Deserialization.parse(parameters)
```

Comment:
One solution to fixing this would be to send `parameters.to_unsafe_h` instead of only `parameters`. Or to try to call that method before calling `to_h` inside the `#parse` method.